src/server/repositories/user.repository.ts

```diff
@@ -0,0 +1,144 @@
+import Product from "../models/Product.models";
+import User from "../models/User.models";
+
+const getOneById = async (id: number) => {
+  const user = await User.findOne({
+    where: {
+      id,
+    },
+    attributes: ["id", "email", "name", "age"],
+    include: {
+      model: Product,
+      attributes: ["id", "title", "price"],
+    },
+    order: [
+      ["id", "DESC"],
+      [Product, "title", "ASC"],
+      [Product, "id", "ASC"],
+    ],
+  });
+  if (user) {
+    return user;
+  }
+
+  return null;
+};
+const getAll = async () => {
+  const users = await User.findAll({
+    attributes: ["id", "email", "name", "age"],
+    include: {
+      model: Product,
+      attributes: ["id", "title", "price"],
+    },
+    order: [
+      ["id", "DESC"],
+      [Product, "title", "ASC"],
+      [Product, "id", "ASC"],
+    ],
+  });
+  return users;
+};
+
+const findByIdAndUpdate = async (id: number, name: string, age: number) => {
+  const countUpdated = await User.update(
+    { name, age },
+    {
+      where: {
+        id,
+      },
+    }
+  );
+  if (countUpdated[0] > 0) {
+    return true;
+  }
+
+  return false;
+};
+
+const findByIdAndDelete = async (id: number) => {
+  const countDeleted = await User.destroy({
+    where: {
+      id,
+    },
+  });
+
+  if (countDeleted > 0) {
```

Comment:
you can call `return countDeleted > 0` instead. Please check all the repositories.